pkg/volume/flexvolume/probe_test.go

```diff
@@ -0,0 +1,294 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flexvolume
+
+import (
+	"fmt"
+	"path"
+	"testing"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/stretchr/testify/assert"
+	utilfs "k8s.io/kubernetes/pkg/util/filesystem"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+const (
+	pluginDir  = "/flexvolume"
+	driverName = "fake-driver"
+)
+
+// Probes a driver installed before prober initialization.
+func TestProberExistingDriverBeforeInit(t *testing.T) {
+	// Arrange
+	driverPath, _, watcher, prober := initTestEnvironment(t)
+
+	// Act
+	updated, plugins, err := prober.Probe()
+
+	// Assert
+	assert.True(t, updated)
+	assert.Equal(t, 1, len(plugins))
+	assert.Equal(t, pluginDir, watcher.watches[0])
+	assert.Equal(t, driverPath, watcher.watches[1])
+	assert.NoError(t, err)
```

Comment:
Comment these to make it more clear which semantics you're asserting.

Reply:
And the rest of the asserts too.

Reply:
ACK. Addressing this partially. Adding comments where it's unclear why the assertion can be made.